File: Java/src/main/java/QuestionBank/Solution001_100/Solution55.java
package QuestionBank.Solution001_100;

/**
 * @author windows
 */
public class Solution55 {
    public boolean canJump(int[] nums) {
        int max = 0;
        for (int i = 0; i < nums.length; i++) {
            if (max >= i){
                if (i + nums[i] > max) {
                    max = i + nums[i];
                }
            }
        }
        return max > nums.length - 1;
    }


    public static void main(String[] args) {
        Solution55 solution55 = new Solution55();
        System.out.println(solution55.canJump(new int[]{2, 3, 1, 1, 4}));
    }
}

File: C++/src/Solution1_100/Solution55.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution55.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

bool canJump(vector<int>& nums) {
  int maxR = nums[0];
  int n = nums.size();
  for(int i = 1; i < n; ++i){
    if (i <= maxR){
      maxR = max(maxR, i + nums[i]);
    }
  }
  return maxR >= n - 1;
}
